sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -424,19 +429,19 @@
      * @throws HttpResponseException If the {@code} key is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ConfigurationSetting>> getSettingWithResponse(ConfigurationSetting setting) {
-        return withContext(context -> getSetting(setting, context));
+    public Mono<Response<ConfigurationSetting>> getSettingWithResponse(String key, String label, OffsetDateTime dateTime) {
```

Comment:
Should this and the other method that takes the same parameters be merged into a single API method?

Reply:
```java
  public Mono<Response<ConfigurationSetting>> getSettingWithResponse(String key, String label, OffsetDateTime dateTime) 
  public Mono<ConfigurationSetting> getSetting(String key, String label, OffsetDateTime dateTime)
```
Is that the methods you are refering to? If yes, those two methods are different used. With Response return type has extra infornation than just value.